### PR TITLE
Chore kindergarten layout

### PR DESCRIPTION
**Pull request by fedescarpa**
Lightbulb assets are placeholders, are not definiteve assets.

### CODE REVIEW

File: app/assets/javascripts/mumuki_laboratory/application/kindergarten.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        mumuki.kids.submitButton.disable();
+        mumuki.kids._getResultsAbortedModal().modal();
+        mumuki.presenterCharacter.playAnimation('failure', mumuki.kids._getCharacterImage());
+      }
+    }
+  };
+
+  $(document).ready(() => {
+
+    if ($('.mu-kindergarten').get(0)) {
```
Comment (luchotc) [inline app/assets/javascripts/mumuki_laboratory/application/kindergarten.js:131]: We should extract this to a method!

File: app/views/layouts/_kindergarten.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<div class="mu-kindergarten-header">
+  <div class="mu-kindergarten-character">
+    <img class="mu-kids-character-animation" alt="">
+    <div class="mu-kindergarten-character-speech-bubble">
+      <span class="mu-kindergarten-exercise-description"><%= exercise.description_task %></span>
+      <div class="mu-kindergarten-buttons">
+        <button class="mu-kindergarten-show-context" onclick="mumuki.kindergarten.context.showContext()">
+          <img class="mu-kindergarten-button-image" src="https://mumuki.io/static/show-context.svg" alt="">
```
Comment (luchotc) [inline app/views/layouts/_kindergarten.html.erb:8]: Let's avoid this kind of references. We should upload them as part of the app!

**Review by luchotc — APPROVED** [review]
Just two minor details.
Beautiful @fedescarpa !

File: app/views/layouts/modals/_kindergarten_results_aborted.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<div class="modal fade mu-kids-results-aborted mu-kindergarten-context" id="kids-results-aborted" tabindex="-1" role="dialog" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content danger">
+      <button type="button" class="mu-kindergarten-modal-button mu-close" data-dismiss="modal" aria-label="Close">
+            <span aria-hidden="true">
+              <i class="fa fa-fw fa-times fa-3x"></i>
+            </span>
+      </button>
+      <h4 class="modal-header text-broken">
+        <strong><%= t :errored %></strong>
```
Comment (flbulgarelli) [inline app/views/layouts/modals/_kindergarten_results_aborted.html.erb:10]: :warning: There are multiple issues with these file: 

1. It is called `aborted`, but it has the `errored` state hardcoded
2. It has the `errored` state and `broken` class, but it is used even when tests fail. 

This leads to the following situation when a test fails normally: 

image

File: app/assets/javascripts/mumuki_laboratory/application/kindergarten.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        mumuki.kids.resultAction.pending = this._showOnFailurePopup;
+      },
+      _showOnSuccessPopup(data) {
+        $('.submission-results').html(data.title_html);
+        mumuki.kids._showOnSuccessPopup(data);
+        $('#kids-results .modal-content').addClass(data.status);
+      },
+      _showOnFailurePopup() {
+        mumuki.kids.submitButton.disable();
+        mumuki.kids._getResultsAbortedModal().modal();
```
Comment (flbulgarelli) [inline app/assets/javascripts/mumuki_laboratory/application/kindergarten.js:123]: :warning:  here we are mixing states:  


```javascript
// we are using a the same failure method in all the three scenarios
        mumuki.kids.resultAction.failed = this._showOnFailurePopup;
        mumuki.kids.resultAction.errored = this._showOnFailurePopup;
        mumuki.kids.resultAction.pending = this._showOnFailurePopup;
```

```
// we are not reading the actual state, so we can not provide proper messages, icons and colors
_showOnFailurePopup() {
...
```


```javascript
 _showOnFailurePopup() {
        mumuki.kids.submitButton.disable();
        // we are calling an aborted modal even when this is not necessary an aborted state
        mumuki.kids._getResultsAbortedModal().modal();
```

**Pull request merged**